bindings/nodejs/src/lib.rs

```diff
@@ -22,94 +22,230 @@ use std::str;
 use chrono::DateTime;
 use chrono::NaiveDateTime;
 use chrono::Utc;
-use opendal::services;
-use futures::prelude::*;
+use futures::{TryStreamExt};
 use napi::bindgen_prelude::*;
 
 #[napi]
-pub struct OperatorFactory {}
+pub struct Memory {}
 
 #[napi]
-impl OperatorFactory {
+impl Memory {
+    #[napi(constructor)]
+    pub fn new() -> Self {
+        Self {}
+    }
+
     #[napi]
-    pub fn memory() -> Result<Operator> {
-        let op = opendal::Operator::create(services::Memory::default())
-            .unwrap()
-            .finish();
+    pub fn build(&self) -> Operator {
+        self.make_operator()
+    }
+}
 
-        Ok(Operator::new(op))
+trait OperatorBuilder {
+    fn make_operator(&self) -> Operator;
+}
+
+impl OperatorBuilder for Memory {
+    fn make_operator(&self) -> Operator {
+        Operator(opendal::Operator::create(opendal::services::Memory::default()).unwrap().finish())
     }
 }
 
-#[allow(dead_code)]
 #[napi]
-pub struct ObjectMeta {
-    pub location: String,
-    pub last_modified: i64,
-    pub size: u32
+pub struct Operator(opendal::Operator);
+
+#[napi]
+impl Operator {
+    #[napi]
+    pub fn object(&self, path: String) -> DataObject {
+        DataObject(self.0.object(&path))
+    }
 }
 
 #[napi]
-pub struct Operator {
-    inner: opendal::Operator
+pub enum ObjectMode {
+    /// FILE means the object has data to read.
+    FILE,
+    /// DIR means the object can be listed.
+    DIR,
+    /// Unknown means we don't know what we can do on this object.
+    Unknown,
 }
 
+#[allow(dead_code)]
 #[napi]
-impl Operator {
-    pub fn new(op: opendal::Operator) -> Self {
-        Self { inner: op }
+pub struct ObjectMetadata(opendal::ObjectMetadata);
+
+#[napi]
+impl ObjectMetadata {
+    /// Mode of this object.
+    #[napi(getter)]
+    pub fn mode(&self) -> ObjectMode {
+        match self.0.mode() {
+            opendal::ObjectMode::DIR => ObjectMode::DIR,
+            opendal::ObjectMode::FILE => ObjectMode::FILE,
+            opendal::ObjectMode::Unknown => ObjectMode::Unknown,
+        }
+    }
+
+    /// Content-Disposition of this object
+    #[napi(getter)]
+    pub fn content_disposition(&self) -> Option<String> {
+        self.0.content_disposition().map(|s| s.to_string())
+    }
+
+    /// Content Length of this object
+    #[napi(getter)]
+    pub fn content_length(&self) -> Option<u32> {
```

Comment:
It's highly possible that content_length exceed u32 and `2^53`. Maybe we should use `BigInt`?

FYI: https://napi.rs/docs/concepts/values.en#bigint

Reply:
> It's highly possible that content_length exceed u32 and 2^53.

Wow. It's necessary to use `BigInt`.